server/channels/store/sqlstore/property_value_store.go

```diff
@@ -0,0 +1,353 @@
+	valueJSON, err := json.Marshal(value.Value)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_value_to_insert_map_marshal_value")
+	}
+	if s.IsBinaryParamEnabled() {
+		valueJSON = AppendBinaryFlag(valueJSON)
+	}
+
+	return map[string]any{
+		"ID":         value.ID,
+		"TargetID":   value.TargetID,
+		"TargetType": value.TargetType,
+		"GroupID":    value.GroupID,
+		"FieldID":    value.FieldID,
+		"Value":      valueJSON,
+		"CreateAt":   value.CreateAt,
+		"UpdateAt":   value.UpdateAt,
+		"DeleteAt":   value.DeleteAt,
+	}, nil
+}
+
+func (s *SqlPropertyValueStore) propertyValueToUpdateMap(value *model.PropertyValue) (map[string]any, error) {
+	valueJSON, err := json.Marshal(value.Value)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_value_to_udpate_map_marshal_value")
+	}
+	if s.IsBinaryParamEnabled() {
+		valueJSON = AppendBinaryFlag(valueJSON)
+	}
+
+	return map[string]any{
+		"Value":    valueJSON,
+		"UpdateAt": value.UpdateAt,
+		"DeleteAt": value.DeleteAt,
+	}, nil
+}
+
+func propertyValuesFromRows(rows *sql.Rows) ([]*model.PropertyValue, error) {
+	results := []*model.PropertyValue{}
+
+	for rows.Next() {
+		var value model.PropertyValue
+		var valueJSON string
+
+		err := rows.Scan(
+			&value.ID,
+			&value.TargetID,
+			&value.TargetType,
+			&value.GroupID,
+			&value.FieldID,
+			&valueJSON,
+			&value.CreateAt,
+			&value.UpdateAt,
+			&value.DeleteAt,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := json.Unmarshal([]byte(valueJSON), &value.Value); err != nil {
+			return nil, errors.Wrap(err, "property_values_from_rows_unmarshal_value")
+		}
+
+		results = append(results, &value)
+	}
+
+	return results, nil
+}
+
+func propertyValueFromRows(rows *sql.Rows) (*model.PropertyValue, error) {
+	values, err := propertyValuesFromRows(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(values) > 0 {
+		return values[0], nil
+	}
+
+	return nil, sql.ErrNoRows
+}
+
+type SqlPropertyValueStore struct {
+	*SqlStore
+}
+
+func newPropertyValueStore(sqlStore *SqlStore) store.PropertyValueStore {
+	return &SqlPropertyValueStore{sqlStore}
+}
+
+func (s *SqlPropertyValueStore) Create(value *model.PropertyValue) (*model.PropertyValue, error) {
+	if value.ID != "" {
+		return nil, store.NewErrInvalidInput("PropertyValue", "id", value.ID)
+	}
+
+	value.PreSave()
+
+	if err := value.IsValid(); err != nil {
+		return nil, errors.Wrap(err, "property_value_create_isvalid")
+	}
+
+	insertMap, err := s.propertyValueToInsertMap(value)
+	if err != nil {
+		return nil, err
+	}
+
+	queryString, args, err := s.getQueryBuilder().
+		Insert("PropertyValues").
+		SetMap(insertMap).
+		ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_value_create_tosql")
+	}
+
+	if _, err := s.GetMaster().Exec(queryString, args...); err != nil {
+		return nil, errors.Wrap(err, "property_value_create_insert")
+	}
+
+	return value, nil
+}
+
+func (s *SqlPropertyValueStore) Get(id string) (*model.PropertyValue, error) {
```

Comment:
nitpick: I wonder if `Get` should rely on `GetMany` to simplify code.

Reply:
We'd have to do some error checking to convert missmatched results into `sql.ErrNoRows`, so we don't remove that much code. 1/5 tho